Rakefile

```diff
@@ -50,20 +52,36 @@ namespace :arm do
     end
   end
 
-  desc 'Regen code for each of the Azure Resource Manager projects'
-  task :regen => :clean_generated do
+  desc 'Regen code for each sdk with all its api versions'
+  task :apiVersions => :clean_generated_api_versions do
+    json = get_config_file
     REGEN_EXCLUDES = ['azure_sdk']
-    each_gem do |dir|
+    each_gem do |dir| # dir corresponds to each azure_mgmt_* folder
       unless REGEN_EXCLUDES.include?(dir.to_s)
         puts "\nGenerating #{dir}\n"
-        md = REGEN_METADATA[dir.to_sym]
         ar_base_command = "#{OS.windows? ? '' : 'mono '} #{REGEN_METADATA[:autorest_loc]}"
```

Comment:
minor: With new autorest with npm one, we may not need to have this conditions `OS.windows? ? '' : 'mono '`

Reply:
that would be nice, we just need to try it on the mac, I haven't yet.